libraries/mariadb_helper.rb

```diff
@@ -30,5 +30,49 @@ def mariadb_service_restart_required?(ip, port, _socket)
       restart = true unless port_open?(ip, port)
       restart
     end
+
+    # Helper to determine if running operating system shipped a package for
+    # mariadb server & client. No galera shipped in any os yet.
+    # @param [String] os_platform Indicate operating system type, e.g. centos
+    # @param [String] os_version Indicate operating system version, e.g. 7.0
+    def os_package_provided?(os_platform, os_version)
+      package_provided = false
+      case os_platform
+      when 'centos', 'redhat'
+        package_provided = true if os_version.to_i == 7
+      when 'fedora'
+        package_provided = true if os_version.to_i >= 19
+      end
+      package_provided
+    end
+
+    # Helper to determine mariadb server service name shipped by native package
+    # If no native package available on this platform, return nil
+    # @param [String] os_platform Indicate operating system type, e.g. centos
+    # @param [String] os_version Indicate operating system version, e.g. 7.0
+    def os_service_name(os_platform, os_version)
+      return nil unless os_package_provided?(os_platform, os_version)
+      service_name = 'mariadb'
+      if os_platform == 'fedora' && os_version.to_i == 19
+        service_name = 'mysqld'
+      end
+      service_name
+    end
+
+    # Helper to determine whether to use os native package
+    # @param [Boolean] prefer_os Indicate whether to prefer os native package
+    # @param [String] os_platform Indicate operating system type, e.g. centos
+    # @param [String] os_version Indicate operating system version, e.g. 7.0
+    def use_os_native_package?(prefer_os, os_platform, os_version)
+      return false unless prefer_os
+      if os_package_provided?(os_platform, os_version)
+        true
+      else
+        Chef::Log.warn 'prefer_os_package detected, but no native mariadb'\
+          " package available on #{os_platform}-#{os_version}."\
+          ' Fall back to use community package.'
+        false
+      end
```

Comment:
In this method, it will check if the target platform has provided native package. If no native package available, it will return false anyway, even the user prefers to use os native package. This logic can gurantee that for those platforms without native packages (e.g. ubuntu), this cookbook will not falesly to use them.